common/chat/create.ts

```diff
@@ -0,0 +1,180 @@
+    const prospectSubcoursesFromConversation = conversation?.custom.prospectSubcourse;
+    let prospectSubcourseIds = [];
+
+    try {
+        prospectSubcourseIds = JSON.parse(prospectSubcoursesFromConversation || '[]');
+    } catch (error) {}
+
+    const updatedProspectSubcourse: number[] = [...prospectSubcourseIds, subcourseId];
+    const returnUpdatedProspectSubcourses = Array.from(new Set(updatedProspectSubcourse));
+
+    const updatedConversation = {
+        id: conversationId,
+        custom: {
+            prospectSubcourse: returnUpdatedProspectSubcourses,
+        },
+    };
+
+    await updateConversation(updatedConversation);
+}
+
+async function updateContactConversation(conversationId: string, conversationInfos: ConversationInfos): Promise<void> {
+    const updatedConversation = {
+        id: conversationId,
+        ...conversationInfos,
+    };
+
+    await updateConversation(updatedConversation);
+}
+
+const getOrCreateGroupConversation = async (participants: User[], subcourseId: number, conversationInfos: ConversationInfos): Promise<Conversation> => {
+    await ensureChatUsersExist(participants);
+
+    const subcourse = await prisma.subcourse.findUniqueOrThrow({
+        where: { id: subcourseId },
+        select: { conversationId: true },
+    });
+
+    if (subcourse.conversationId === null) {
+        const newConversationId = await createConversation(participants, conversationInfos, 'group');
+        const newConversation = await getConversation(newConversationId);
+        await sendSystemMessage(systemMessages.de.groupChat, newConversationId, SystemMessage.FIRST);
+        await prisma.subcourse.update({
+            where: { id: subcourseId },
+            data: { conversationId: newConversationId },
+        });
+        const convertedConversation = convertTJConversation(newConversation);
+
+        return convertedConversation;
+    }
+
+    const subcourseGroupChat = await getConversation(subcourse.conversationId);
+    const convertedSubcourseGroupChatConversation = convertTJConversation(subcourseGroupChat);
+
+    return convertedSubcourseGroupChatConversation;
+};
+
+async function createContactChat(meUser: User, contactUser: User): Promise<string> {
+    const myContacts = await getMyContacts(meUser);
+    const contact = myContacts.find((c) => c.user.userID === contactUser.userID);
+
+    if (!contact) {
+        throw new Error('Chat contact not found');
+    }
+
+    const conversationInfos: ConversationInfos = {
+        custom: {
+            ...(contact.match && { match: { matchId: contact.match.matchId } }),
+            ...(contact.subcourse && { subcourse: [...new Set(contact.subcourse)] }),
+        },
+    };
+
+    const conversation = await getOrCreateOneOnOneConversation([meUser, contactUser], conversationInfos, ContactReason.CONTACT);
+    return conversation.id;
```

Comment:
Generally we could use some logging here.

Reply:
Once we rollout the Chat you have to troubleshoot this, not sure if that is possible without logs ...